shared/entity/type.go

```diff
@@ -89,6 +89,11 @@ func nRequiredPathArguments(t Type) int {
 	return nRequiredPathArguments
 }
 
+// Is returns true if the expected Type is equal to the expected Type.
+func Is(expected Type, actual Type) bool {
+	return expected.Name() == actual.Name()
```

Comment:
Can we use type assertion here to check the concrete type inside the interfaces match for `Type`s passed in?

Reply:
I don't think so. We wouldn't be able to use this function for comparing a `cluster.EntityType` to an `entity.Type` implementation in that case. They are the same kind if the name is equal.